src/use-swr.ts

```diff
@@ -163,13 +174,31 @@ const mutate: mutateInterface = async (
   }
 
   if (typeof data !== 'undefined') {
-    // update cached data, avoid notifying from the cache
+    // update cached data
     cache.set(key, data)
   }
+  // always update or reset the error
   cache.set(keyErr, error)
 
   // reset the timestamp to mark the mutation has ended
-  MUTATION_END_TS[key] = Date.now() - 1
+  MUTATION_END_TS[key] = now() - 1
+
+  if (!isMutationAsync) {
+    // let's always broadcast in the next tick
+    // to dedupe synchronous mutation calls
+    // check out https://github.com/vercel/swr/pull/735 for more details
+    await 0
+
+    // we skip broadcasting if there's another mutation
+    // happened synchronously
```

Comment:
seems are same code with above, maybe we can make it a short function? like

```js
const resolveDataOnMutation = () => {
  if (
      beforeMutationTs !== MUTATION_TS[key] ||
      beforeConcurrentPromisesTs !== CONCURRENT_PROMISES_TS[key]
    ) {
      if (error) throw error
      return data
    }
  }
}
```

you might come up with other better names